src/aliceVision/keyframe/KeyframeSelector.cpp

```diff
@@ -380,28 +453,33 @@ bool KeyframeSelector::computeScores(const std::size_t rescaledWidthSharpness, c
              *   - try reading the next frame instead
              *   - if the next frame is correctly read, then push dummy scores for the invalid frame and go on with
              *     the process
-             *   - otherwise (feed not correctly moved to the next frame), throw a runtime error exception as something
-             *     is wrong with the video
+             *   - otherwise (feed not correctly moved to the next frame), keep on going to the next frame until it is
+             *     valid or the end of the feed is reached
              */
             if (!skipSharpnessComputation) {
                 try {
                     // Read image for sharpness and rescale it if requested
                     currentMatSharpness = readImage(feed, rescaledWidthSharpness);
                 } catch (const std::invalid_argument& ex) {
-                    // currentFrame + 1 = currently evaluated frame with indexing starting at 1, for display reasons
-                    // currentFrame + 2 = next frame to evaluate with indexing starting at 1, for display reasons
-                    ALICEVISION_LOG_WARNING("Invalid or missing frame " << currentFrame + 1
-                                            << ", attempting to read frame " << currentFrame + 2 << ".");
-                    bool success = feed.goToFrame(++currentFrame);
-                    if (success) {
-                        // Will throw an exception if next frame is also invalid
-                        currentMatSharpness = readImage(feed, rescaledWidthSharpness);
-                        // If no exception has been thrown, push dummy scores for the frame that was skipped
-                        _sharpnessScores.push_back(-1.f);
-                        _flowScores.push_back(-1.f);
-                    } else
-                        ALICEVISION_THROW_ERROR("Could not go to frame " << currentFrame + 1
-                                                << " either. The feed might be corrupted.");
+                    bool success = false;
+                    while (!success && currentFrame < nbFrames) {
+                        // currentFrame + 1 = currently evaluated frame with indexing starting at 1, for display reasons
+                        // currentFrame + 2 = next frame to evaluate with indexing starting at 1, for display reasons
+                        ALICEVISION_LOG_WARNING("Invalid or missing frame " << currentFrame + 1
+                                                << ", attempting to read frame " << currentFrame + 2 << ".");
+
+                        {
+                            // Push dummy scores for the frame that was skipped
+                            const std::lock_guard<std::mutex> lock(_mutex);
```

Comment:
`scoped_lock` is preferred since C++17 (same remark below)

Reply:
Thanks, done.